addons/web/static/lib/lanczos/lanczos.js

```diff
@@ -0,0 +1,162 @@
+
+                cx = x * iw - centerX;
+                sum = a = r = g = b = 0;
+                for (y1 = y1s >> 0; y1 <= y1e; y1++) {
+                    srcOffset = y1 * img.width;
+                    distY = (y1 + cy) * (y1 + cy) * sy2;
+                    for (x1 = x1s >> 0; x1 <= x1e; x1++) {
+                        // Weight computation with cache
+                        weight = cache[((x1 + cx) * (x1 + cx) * sx2 + distY) * cachePrecision >> 0] || 0;
+                        sum += weight;
+                        // Color accumulation
+                        idx = (srcOffset + x1) << 2;
+                        r += srcData[idx] * weight;
+                        g += srcData[idx + 1] * weight;
+                        b += srcData[idx + 2] * weight;
+                        a += srcData[idx + 3] * weight;
+                    }
+                }
+                sum = sum > 0 ? 1.0 / sum : 0.0;
+                idx = (dstOffset + x) << 2;
+                dstData[idx] = r * sum;
+                dstData[idx + 1] = g * sum;
+                dstData[idx + 2] = b * sum;
+                dstData[idx + 3] = a * sum;
+            }
+        }
+        const outputCanvas = createCanvas(dst.width, dst.height);
+        const outputCtx = outputCanvas.getContext("2d");
+        outputCtx.putImageData(dst, 0, 0)
+        return outputCanvas;
+    }
+
+    function resizeLanczos(img, newWidth, newHeight, options = {}) {
+        // Avoid decimal sizes
+        const newW = Math.floor(newWidth);
+        const newH = Math.floor(newHeight);
+        const canvas = resampleLanczos(img, newW, newH, options);
+        return canvas;
+    }
+
+    function resizeCanvas(image, newWidth, newHeight) {
+        const canvas = createCanvas(newWidth, newHeight);
+        const ctx = canvas.getContext("2d");
+        ctx.fillStyle = "transparent";
+        ctx.fillRect(0, 0, canvas.width, canvas.height);
+        ctx.imageSmoothingEnabled = true;
+        ctx.imageSmoothingQuality = "high";
+        ctx.drawImage(
+            image,
+            0, 0, image.width, image.height,
+            0, 0, canvas.width, canvas.height
+        );
+        return canvas;
+    }
+
+    function resizeImageWithResampling(image, newSize, originalSize, options = {}) {
+        const ratio = newSize / originalSize;
+        const newWidth = image.width * ratio;
+        const newHeight = image.height * ratio;
+        // Keep it thumbnails only for now to avoid performance issues
+        if (newSize < originalSize && newSize < 1920) {
+            try {
+                return resizeLanczos(image, newWidth, newHeight, options);
+            } catch (error) {
+                console.warn("Lanczos resizing failed, falling back to canvas resizing:", error);
+            }
+        }
+        return resizeCanvas(image, newWidth, newHeight);
+    }
+
+    exports.resizeImageWithResampling = resizeImageWithResampling;
+
+})(this);
```

Comment:
scope the lib into window.lanczos

```suggestion
})(this.lanczos = this.lanczos || {});
```

Then, use it this way:
```js
lanczos.resizeImageWithResampling(...);
```